src/ivalue_manager.rs

```diff
@@ -569,6 +585,25 @@ impl ReadHolder<IValue> for IValueKeyHolderRead {
     }
 }
 
+fn merge(doc: &mut IValue, patch: &IValue) {
+    if !patch.is_object() {
+        *doc = patch.clone();
+        return;
+    }
+
+    if !doc.is_object() {
```

Comment:
If doc is not an object, it will be completely overwritten (replaced) by the patch value? (merge takes all from the right-hand-side patch object, and nothing from the left-hand-side array or scalar)

Reply:
yes, I didn't do the arrays support yet

Reply:
I think this is what we agreed on merge arrays, right? @adrianoamaral ?

Reply:
yes, following the RFC7396 - array, replaced by the new value. Other behaviours we can handle on the JSON.ARRAPPEND